### PR TITLE
Add more options to Service Products

### PR DESCRIPTION
**Pull request by dhafer95**

### CODE REVIEW

File: src/views/components/product/options/map.twig [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<div class="map-wrapper map-wrapper--has-search">
+  <button id="current_location_button_[{{ option.id }}]" type="button" class="btn map-wrapper__current-button">
+    <i class="sicon-location-target"></i>
+  </button>
+  <input id="map_search_input_[{{ option.id }}]" class="form-control map-wrapper__map-search" type="text" placeholder="بحث عن موقع">
```
Comment (mshalan) [inline src/views/components/product/options/map.twig:5]: need to add translations

File: src/views/pages/customer/orders/single.twig [hunk 2]
```diff
@@ -212,8 +214,33 @@
...
+                                          {% elseif option.is_file %}
+                                            <salla-button shape="link" class="s-button-wrap hydrated" color="primary" fill="solid" size="medium" width="normal">
+                                              <a href="{{ option.value }}" target="_blank" class="font-medium text-xs underline">ملف مرفق</a>
+                                            </salla-button>
+                                          {% elseif option.is_color_picker %}
+                                            <span class="inline-block h-6 w-6 ms-2 rounded-full shadow" style="background-color: {{ option.value }}" ></span>
+                                          {% elseif option.is_map %}
+                                            <salla-button shape="link" onclick="document.querySelector('#location-modal-{{option.id}}').open()" class="s-button-wrap hydrated" color="primary" fill="solid" size="medium" width="normal">
+                                              <a class="font-medium text-xs underline">عرض الموقع</a>
```
Comment (mshalan) [inline src/views/pages/customer/orders/single.twig:225]: need to add translations

File: src/views/pages/customer/orders/single.twig [hunk 3]
```diff
@@ -212,8 +214,33 @@
                                               </div>
                                           {% elseif option.color %}
                                             <span title="{{option.value}}" class="inline-block h-6 w-6 ms-2 rounded-full shadow" style="background-color: {{ option.color }}" ></span>
+                                          {% elseif option.is_file %}
+                                            <salla-button shape="link" class="s-button-wrap hydrated" color="primary" fill="solid" size="medium" width="normal">
+                                              <a href="{{ option.value }}" target="_blank" class="font-medium text-xs underline">ملف مرفق</a>
```
Comment (mshalan) [inline src/views/pages/customer/orders/single.twig:219]: need to add translations

**Review by mshalan — CHANGES_REQUESTED** [review]

File: src/views/components/product/options/map.twig (hunk 1 above)
Comment (ossycodes) [reply src/views/components/product/options/map.twig]: @mshalan resolved

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

File: src/views/pages/customer/orders/single.twig [hunk 4]
```diff
@@ -207,48 +207,75 @@
...
                                             </div>
-                                        {% endfor %}
-                                    </dl>
-                                </div>
-                            {% endif %}
+                                          {% elseif option.is_map %}
+                                            <salla-map
+                                                id="location_map"
+                                                class="map_{{option.id}}"
+                                                api-key="{{ store.settings.google_map_key }}"
```
Comment (jalmatari) [inline src/views/pages/customer/orders/single.twig:258]: no need for this

File: src/views/layouts/master.twig [hunk 5]
```diff
@@ -27,7 +27,8 @@
 | store.settings.rating_enabled                      | bool     | Is one of: store_enabled or shipping_enabled or products_enabled                                         |
 | store.settings.arabic_numbers_enabled              | bool     | Does the merchant wants the numbers to be shown as Arabic format                                         |
 | store.settings.is_multilingual                     | bool     |                                                                                                          |
-| store.settings.currencies_enabled                  | bool     |                                                                                                          |
+| store.settings.currencies_enabled                  | bool     |     
+| store.settings.google_map_key                      | string   |                                                                           |
```
Comment (jalmatari) [inline src/views/layouts/master.twig:31]: no need for this line, we are passing the key directly via the twilight event

**Review by jalmatari — CHANGES_REQUESTED** [review]

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

File: src/views/pages/customer/orders/single.twig (hunk 4 above)
Comment (ossycodes) [reply src/views/pages/customer/orders/single.twig:258]: @dhafer95 @eslamoo

File: src/views/layouts/master.twig (hunk 5 above)
Comment (ossycodes) [reply src/views/layouts/master.twig:31]: @dhafer95 @eslamoo

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has been synced from master..✅

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
> The branch `feature-more-options-service` of https://github.com/SallaApp/theme-raed/pull/83 has Merge conflict❌

**Comment by SallaDev** [comment]
https://github.com/SallaApp/theme-raed/pull/83 ` is been merged ✅`

-------------------

**Pull request merged**